### PR TITLE
Konosuba TXPL support

### PR DESCRIPTION
**Pull request by nikvoid**
LZLR Compression (currently simple and uneffective compression only), TexPool (.txl and .tlz) extracting and building, Tex encoding

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -454,13 +501,113 @@ private static int TexDecode(ReadOnlySpan<string> args)
...
+            if (args.Length < 2)
+            {
+                Console.Error.WriteLine("Usage: ShinDataUtil tex-encode [inpng] [outtex] [--format] [--lzlr]");
+                Console.Error.WriteLine("--lzlr     compress output with LZLR");
+                Console.Error.WriteLine("--format   specify out texture format");
+                Console.Error.WriteLine($"           default: {format}");
+                Console.Error.WriteLine("           available formats:");
+                foreach (var fmt in NVNTexture.GetAvailableFormats())
+                {
+                Console.Error.WriteLine($"           {(uint)fmt}  {fmt}");
```
Comment (DCNick3) [inline Program.cs]: nit: add tab

File: Program.cs [hunk 2]
```diff
@@ -454,13 +501,113 @@ private static int TexDecode(ReadOnlySpan<string> args)
...
+                
+                return 1;
+            }
+
+            var inpng = args[0];
+            var outtex = args[1];
+
+            if (args.Contains("--format"))
+            {
+                Int32.TryParse(args[args.IndexOf("--format") + 1], out var fmt);
```
Comment (DCNick3) [inline Program.cs]: I think using `Enum.TryParse` to parse symbolic names for the formats would be better for the user. ~~Also please check whether the parse was successful and raise an error if it was not =)~~ Scratch that, the enum is checked against the list from `GetAvailableFormats`, that's fine

**Review by DCNick3 — CHANGES_REQUESTED** [review]
Overall - seems good, thanks for taking time to upstream =)

File: Program.cs (hunk 1 above)
Comment (nikvoid) [reply Program.cs]: I decided to leave it without tab intentionally, just to make it look like in console output
P.S. можем на русском)

Comment (DCNick3) [reply Program.cs]: Справедливо (хотя форматтер это убьёт скорее всего когда-нибудь 😅)

Comment (nikvoid) [reply Program.cs]: Возможно...

File: Program.cs (hunk 2 above)
Comment (nikvoid) [reply Program.cs]: Сделаю тогда ввод имени формата вместо номера

**Comment by DCNick3** [comment]
Норм

**Pull request merged**